src/core/control/tools/StrokeHandler.cpp

```diff
@@ -266,9 +267,20 @@ void StrokeHandler::onButtonReleaseEvent(const PositionInputData& pos, double zo
     Layer* layer = page->getSelectedLayer();
 
     UndoRedoHandler* undo = control->getUndoRedoHandler();
-
     undo->addUndoAction(std::make_unique<InsertUndoAction>(page, layer, stroke.get()));
 
+    // If stroke was on last page, ensure there is always an empty last page by creating one (if configured)
+    auto* doc = control->getDocument();
+    if (!doc->isAttachPdf() && settings->getAddEmptyLastPage()) {
```

Comment:
`isAttachPdf()` only returns `true` if the PDF is attached to the document (which is usually not the case when annotating a PDF). It's better to check if there are pdf pages:
```suggestion
    if (doc->getPdfPageCount() == 0 && settings->getAddEmptyLastPage()) {
```